pkg/harvester/api/api.go

```diff
@@ -30,18 +34,28 @@ func NewHTTPApi(controller controller.HarvesterController) API {
 
 func (a *HTTPApi) Run(ctx context.Context) error {
 	a.logger.Info("Starting HTTP API")
-	// TODO: implement
 
-	<-ctx.Done()
+	var controller controller.HarvesterController
+
+	harvester_api := NewHTTPApi(controller)
+
+	err := a.run(ctx, harvester_api)
+	if err != nil {
+		a.logger.Error("HTTP API has failed", err)
+		panic(err)
```

Comment:
could you try if this works ok? 
```suggestion
		return fmt.Errorf("HTTP API has failed: %v", err)
```

Reply:
of course! but why are we not logging the error ? I mean, is this `log.Logger` adding logs to any file or something like that ? or does it just prints out the output in the terminal/cmd ?